runtime/tests/third_party/dummy_device.toml

```diff
@@ -91,6 +58,38 @@ matrix = [
         "MultiControlledX",
 ]
 
+[operators.gates.named]
+
+QubitUnitary = { controllable = true }
```

Comment:
This format looks good to me 👍 I would also add `invertible` and `differentiable` flags

Reply:
For the invertible flag - it is the same to adjoint, right? We currently check it in one place [here](https://github.com/PennyLaneAI/catalyst/blob/3b27b055d38b9e89ecaf2acfcdeabd82ad75c14c/frontend/catalyst/qjit_device.py#L122), and the action is to add the 'Adjoint' operation to the set of supported ops. What would per-gate flag do?

The same is for `differentiable`. Do you have a concrete suggestion how to check/validate it?

Reply:
> We currently check it in one place [here](https://github.com/PennyLaneAI/catalyst/blob/3b27b055d38b9e89ecaf2acfcdeabd82ad75c14c/frontend/catalyst/qjit_device.py#L122), and the action is to add the 'Adjoint' operation to the set of supported ops. What would per-gate flag do?

Hmm it's true that on our side how we interpret adjoint support is to add the `Adjoint` op to the list of supported gates in the QJITDevice, so setting the flag on a per operation gate doesn't do much at that level. However, for verifying that a program is supported we could use the per-gate flag to ensure all ops on the tapes nested inside `Adjoint` are part of the supported set.

For the lightning devices I believe all native gates support inverting, so having the per-gate flag is not strictly necessary yet. However, that's how we envisioned the control modifier initially as well and it turned out more fine-grained control was needed. So I think being proactive here is not a bad idea, rather than waiting for the next device that will need a fine-grained adjoint spec and update the schema again at that point.

Now for _convenience_ I could also see the benefit of having a global flag, but I think it should be in addition to the gate-level flags. It would just prevent TOML file authors from specifying the flag repeatedly when they want it to apply to all gates anyways.

Reply:
> The same is for differentiable. Do you have a concrete suggestion how to check/validate it?

For differentiability we would simply check whether the final circuit contains any gates not listed with differentiable support. This would only apply to circuits inside of `grad` calls though. And this would only apply to QNodes with `diff_method=adjoint` since at least for now that is the only device based differentiation supported. 
For compiler-based differentiation methods we would have a different verifier not based on the TOML information.

Reply:
Note that full verification doesn't have to be part of this PR yet, we can implement it in subsequent PRs but the relevant information needs to be in-place already.

Reply:
Here, let me state, that I see our TOML API as follows:
```
get_decomposition_spec : TOMLDocument -> DeviceOptions -> DecompositionSpec
validate_program : TOMLDocument -> DeviceOptions -> Program -> bool
```
where
* by `DeviceOptions` I mean user device settings like `shots=XXX, wires=YYY`
* by `DecompositionSpec` I mean PL's `QubitDevice.operations` supporting the `Gate, C(Gate), Adjoint(Gate)` syntax. 

In these terms, this PR addresses the first API method, encoded here as [get_pennylane_operations](https://github.com/PennyLaneAI/catalyst/blob/9cb6e3d44c6d83106dc23ffe0559af55caafc331/frontend/catalyst/utils/runtime.py#L92). 
Why do we want to check the decomposition specs? Because it is encoded (at least) twice, first time as a toml file, second time as `QubitDeviec.operations` set by the PL directly. Here is [how it looks](https://github.com/XanaduAI/sergei-environment/blob/master/src/qcontrol/toml.md#checking-the-pl-device-operations) btw.

Regarding the invertibility, while PL spec includes some `Adjoint(...)` gates, the meaning is not clear to me. I assume that all Catalyst-compatible devices support invertibility for all gates, and that is why we don't check them (we just filter them out).

The differentiability in my understanding is related to the second `validate_program` method, which is yet to be developed.

So, while I am agree with your requirements (adding them as gate options) I would suggest to postpone introducing both until we can write a meaningful processing.